back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -114,17 +113,6 @@ def raise_error_if_stale_form_data
       raise UpdateAllFailedError, { form: [{ error: 'stale_data' }] }
     end
 
-    def raise_error_if_no_end_page
-      pmethod = @custom_form.participation_context.pmethod.class.method_str
-      is_survey_or_ideation = pmethod.in?(%w[native_survey ideation])
-      return unless is_survey_or_ideation
-
-      last_field = update_all_params.fetch(:custom_fields).last
-      return if last_field['key'] == 'form_end' && last_field['input_type'] == 'page'
-
-      raise UpdateAllFailedError, { form: [{ error: 'no_end_page' }] }
-    end
-
```

Comment:
Noticed that there was duplication here with a method we're already calling a few lines later - `IdeaCustomFieldService.check_form_structure`, so just made the latter a little more robust